modules/sc-mesh-secure-deployment/src/nats/comms_nats_controller.py

```diff
@@ -1155,10 +1156,57 @@ def __cbma_process_alive_check(self):
         for process_name in terminated_processes:
             del self.__cbma_processes[process_name]
 
+    def __validate_response(
+        self, response: requests.Response, config: ConfigType, status_type: StatusType
+    ) -> str:
+        """
+        Validate response
+        :param response: HTTP response
+        :param config: config
+        :param status_type: status type
+        """
+        # validation for response parameters
+
+        self.__comms_controller.logger.debug(
+            "validating response: %s, Config %s", response.text.strip(), config
+        )
+        status: str = "FAIL"
+
+        if response.status_code == 200:
+            if config == ConfigType.FEATURES:
+                try:
+                    if json.loads(response.text)["payload"]["features"]:
+                        status = "OK"
+                except KeyError:
+                    self.__comms_controller.logger.error(
+                        "Features field not found in config"
+                    )
+            elif config == ConfigType.MESH_CONFIG:
+                try:
+                    if json.loads(response.text)["payload"]["radios"]:
+                        status = "OK"
+                except KeyError:
+                    self.__comms_controller.logger.error(
+                        "Radios field not found in config"
+                    )
+            elif config == ConfigType.DEBUG_CONFIG:
```

Comment:
Maybe for completeness and robustness add an `else` after this `elif` that fails with 'unknown config' or something catastrophically beautiful :fairy:

Reply:
[34fcb4a](https://github.com/tiiuae/mesh_com/pull/389/commits/34fcb4a106727a4f90a0c8bc7a3291406f16aea1)